girder/stemserver_plugin/models/stemimage.py

```diff
@@ -320,32 +310,40 @@ def _get_vlen_dataset_in_chunks(self, dataset, limit, offset,
 
         yield data
 
-    def _get_dataset_in_chunks(self, dataset, max_chunk_size=64000):
+    def _get_dataset_in_chunks(self, dataset, array_position=None, max_chunk_size=64000):
         """A generator to yield numpy arrays of the dataset.
 
         Args:
             dataset: An h5py dataset
+            array_position: The index of the array in the dataset to be
+                            yielded. If array_position is None, yield all
+                            the arrays in the dataset
             max_chunk_size: The maximum size in bytes to be sent. This
                             will be used to determine how many arrays
                             to send. Note that it will always send at
                             least one array, even if the size exceeds
                             the max.
         Yields: Arrays of the dataset
         """
-        approx_items_per_array = dataset.size / dataset.shape[0]
-        approx_size_per_array = approx_items_per_array * dataset.dtype.itemsize
+        if array_position is None:
+            approx_items_per_array = dataset.size / dataset.shape[0]
+            approx_size_per_array = approx_items_per_array * dataset.dtype.itemsize
 
-        read_size = int(max_chunk_size // approx_size_per_array)
-        if read_size == 0:
-            read_size = 1
+            read_size = int(max_chunk_size // approx_size_per_array)
+            if read_size == 0:
+                read_size = 1
 
-        if read_size > dataset.shape[0]:
-            read_size = dataset.shape[0]
+            if read_size > dataset.shape[0]:
+                read_size = dataset.shape[0]
+            read_limit = dataset.shape[0]
+        else:
+            read_size = 1
+            read_limit = 1
```

Comment:
Is the `array_position` supposed to be used as an index somewhere? I don't see it being used other than checking if it is `None`.